src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,67 @@
+package duke.storage;
+
+import duke.DukeException;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public List<Task> load() throws DukeException {
+        List<Task> loadedTasks = new ArrayList<>();
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(filePath));
+            for (String line : lines) {
+                String[] parts = line.split(" \\| ");
+                Task task = null;
+                switch (parts[0]) {
+                    case "T":
+                        task = new Todo(parts[2], parts[1].equals("1"));
+                        break;
```

Comment:
Perhaps intead of using "parts" you can be more specific as to what it refers to e.g. "eventDetails" , "deadlineDetails", "eventTiming"